src/snowflake/cli/_plugins/nativeapp/manager.py

```diff
@@ -591,8 +593,7 @@ def _expand_script_templates(
         scripts_contents = []
         for relpath in scripts:
             try:
-                template = env.get_template(relpath)
-                result = template.render(**jinja_context)
+                result = render_from_file(relpath, jinja_context)
```

Comment:
Passing this function in feels like a hack. Sometimes you have to pass the full path, sometimes you have to pass relative path, and it all depends on what project root it's assumed there.

Some suggestions:
Unify the interface for both {{ }} and SQL rendering, maybe through abstract methods and ABC  classes.
We could have something like 2 ABC classes, one for files, and one for string, with render_from_file and render_from_string methods respectively. Then we can have custom classes that can implement different interfaces:
SqlTemplateEnv(FileTemplateEnv, StringTemplateEnv)
CustomTemplateEnv(FileTemplateEnv)
GenericTemplateEnv(FileTemplateEnv)

Then in _expand_script_templates() we can pass in FileTemplateEnv as an argument.

Reply:
IMO this would overcomplicate rendering interface only to avoid code duplication above. I changed the approach - now both `jinja` and `sql` modules have `render_from_str` functions, which require file contents instead of a path, and above function require paths relative to the project root. It unwraps the files and handles appropriate exceptions itself, without relying on rendering interface.